apps/src/templates/UnassignButton.jsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {connect} from 'react-redux';
+import Button from './Button';
+import i18n from '@cdo/locale';
+import {unassignSection} from '@cdo/apps/templates/teacherDashboard/teacherSectionsRedux';
+
+class UnassignButton extends React.Component {
+  static propTypes = {
+    sectionId: PropTypes.number.isRequired,
+    // Redux
+    unassignSection: PropTypes.func.isRequired
+  };
+
+  constructor() {
+    super();
+    this.state = {text: i18n.assigned(), icon: 'check'};
+  }
+
+  onMouseOver = event => {
+    this.setState({text: i18n.unassign(), icon: 'times'});
+  };
+
+  onMouseOut = event => {
+    this.setState({text: i18n.assigned(), icon: 'check'});
```

Comment:
it would be a bit more conventional to store `isMouseOver` or similar boolean in the state, and then set the text and icon based on that in the render method below

Reply:
good idea, I'll address in follow-up.